### PR TITLE
Add skipDockerPush and skipDocker properties

### PR DESCRIPTION
**Pull request by cthiebault**
Add
* `skipDocker` to skip any Docker goal (useful when goals are bound to maven phases)
* `skipDockerPush` to skip image push when `push` goal is bound to `deploy` phase

This PR is following https://github.com/spotify/docker-maven-plugin/issues/203

### CODE REVIEW

**Comment by davidxia** [comment]
@cthiebault Thanks for your PR. In addition to `PushMojo`, `BuildMojo` and `TagMojo` also call `pushImage()`. Could we add a test for this? See examples in `PushMojoTest`.

**Comment by cthiebault** [comment]
No problem, I'll do that tomorrow!

**Comment by cthiebault** [comment]
I've added some tests for the skip actions.
I've also added `slf4j-simple` for tests so we don't have `Failed to load class org.slf4j.impl.StaticLoggerBinder` error message anymore:
```
<dependency>
  <groupId>org.slf4j</groupId>
  <artifactId>slf4j-simple</artifactId>
  <version>1.7.19</version>
  <scope>test</scope>
</dependency>
```

**Comment by cthiebault** [comment]
Any news on this PR? I'd like to use these skip flags...
Maybe I've changed too many files? Do you prefer I reduce the scope of my commits?

**Comment by guilhox** [comment]
Hi! Is this PR still being looked into? We have a team eager to use those flags as well...

**Comment by guzmo** [comment]
Same here :)

File: src/test/java/com/spotify/docker/BuildMojoTest.java [hunk 1]
```diff
@@ -53,13 +53,10 @@
...
+import static com.spotify.docker.TestUtils.getPomAndAssertExists;
 import static java.nio.charset.StandardCharsets.UTF_8;
-import static org.mockito.Matchers.any;
-import static org.mockito.Matchers.anyString;
-import static org.mockito.Matchers.eq;
-import static org.mockito.Mockito.doAnswer;
-import static org.mockito.Mockito.mock;
-import static org.mockito.Mockito.verify;
+import static org.assertj.core.api.Assertions.*;
+import static org.mockito.Mockito.*;
```
Comment (mattnworb) [inline src/test/java/com/spotify/docker/BuildMojoTest.java]: would prefer to not use star imports

File: src/test/java/com/spotify/docker/TestUtils.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class TestUtils {
+
+  private TestUtils() {
+  }
+
+  public static File getPomAndAssertExists(String filename) throws Exception {
+    final File pom = new File(TestUtils.class.getResource(filename).toURI());
+    assertThat(pom).isNotNull().exists();
```
Comment (mattnworb) [inline src/test/java/com/spotify/docker/TestUtils.java]: very minor but: `pom` can never be null

**Comment by mattnworb** [comment]
LGTM overall, and thanks for the contribution @cthiebault. Looks like there are merge conflicts though. Would be happy to merge if you could address the star import thing above.

**Comment by cthiebault** [comment]
I've just applied your comments and fixed the conflicts...

**Comment by mattnworb** [comment]
@cthiebault thanks again!

**Comment by derekbassett** [comment]
This is SO awesome is there any way we can get a new drop of the plugin?  I have a delivery for some internal software and would prefer not to fork our instance of the plugin if possible.

**Comment by davidxia** [comment]
@derekbassett I just released 0.4.12.

**Pull request merged**